### PR TITLE
OHRM5X-1300: Admin User Creation for Installer

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-1300: [Installer][FE] Admin User Creation

### CODE REVIEW

File: installer/client/src/pages/AdminUserCreation.vue [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+          <oxd-input-field
+            v-model="adminUser.firstName"
+            :rules="rules.firstName"
+          />
+          <oxd-input-field
+            v-model="adminUser.lastName"
+            :rules="rules.lastName"
+          />
+        </oxd-input-group>
+      </oxd-grid-item>
```
Comment (devishke-orange) [inline installer/client/src/pages/AdminUserCreation.vue:47]: Need help with styles for this. Currently appearing like this:

image

Need to look like this:
image

Comment (RajithaKumara) [reply installer/client/src/pages/AdminUserCreation.vue:47]: try nested grid:
```vue
<oxd-grid :cols="2" class="orangehrm-full-width-grid">
    <oxd-grid-item class="--span-column-2">
        <oxd-input-group label="Employee Name" :classes="classes">
          <oxd-input-field
            v-model="adminUser.firstName"
            :rules="rules.firstName"
          />
      </oxd-grid-item>
    <oxd-grid-item class="--span-column-2">
          <oxd-input-field
            v-model="adminUser.lastName"
            :rules="rules.lastName"
          />
        </oxd-input-group>
      </oxd-grid-item>
</oxd-grid>
```
you may have adjust `:cols="2" class="orangehrm-full-width-grid"`
`class="--span-column-2"`

Comment (devishke-orange) [reply installer/client/src/pages/AdminUserCreation.vue:47]: tried with a nested grid and some other variations but couldn't get it to work :frowning_face:

File: installer/client/src/pages/AdminUserCreation.vue [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        confirmPassword: '',
+      },
+      rules: {
+        firstName: [required, shouldNotExceedCharLength(30)],
+        lastName: [required, shouldNotExceedCharLength(30)],
+        email: [
+          required,
+          (v) =>
+            !v ||
+            /^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)+$/.test(
```
[the remark this answers is not in the document]
Comment (devishke-orange) [reply installer/client/src/pages/AdminUserCreation.vue]: done

File: installer/client/src/pages/AdminUserCreation.vue [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+              v,
+            ) ||
+            'Expected format: admin@example.com',
+        ],
+        contact: [
+          shouldNotExceedCharLength(25),
+          (v) =>
+            !v ||
+            /^[0-9+\-/() ]+$/.test(v) ||
```
[the remark this answers is not in the document]
Comment (devishke-orange) [reply installer/client/src/pages/AdminUserCreation.vue]: moved to installer rules

File: installer/client/src/pages/AdminUserCreation.vue (hunk 1 above)
Comment (devishke-orange) [reply installer/client/src/pages/AdminUserCreation.vue]: got it working with a nested grid not sure if the approach I took is okay. if you compare between the name and password fields you can see it is still a little off.

image

File: installer/client/src/pages/AdminUserCreation.vue [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        .create({
+          firstName: this.adminUser.firstName,
+          lastName: this.adminUser.lastName,
+          email: this.adminUser.email,
+          contact: this.adminUser.contact,
+          username: this.adminUser.username,
+          password: this.adminUser.password,
+        })
+        .then(() => {
+          // TODO navigate to confirmation page
```
Comment (RajithaKumara) [inline installer/client/src/pages/AdminUserCreation.vue:189]: *

File: installer/client/src/pages/AdminUserCreation.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        });
+    },
+    navigateUrl() {
+      // TODO navigate to instance creation page
```
Comment (RajithaKumara) [inline installer/client/src/pages/AdminUserCreation.vue:193]: *

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**